src/uct/ib/rc/accel/rc_mlx5_common.c

```diff
@@ -630,17 +649,48 @@ static void uct_rc_mlx5_iface_common_dm_tl_cleanup(uct_mlx5_dm_data_t *data)
 
 #if IBV_HW_TM
 void uct_rc_mlx5_init_rx_tm_common(uct_rc_mlx5_iface_common_t *iface,
+                                   const uct_rc_iface_common_config_t *config,
                                    unsigned rndv_hdr_len)
 {
     uct_ib_md_t *md       = uct_ib_iface_md(&iface->super.super);
     unsigned tmh_hdrs_len = sizeof(struct ibv_tmh) + rndv_hdr_len;
+    ucs_status_t status;
 
     iface->tm.eager_desc.super.cb = uct_rc_mlx5_release_desc;
-    iface->tm.eager_desc.offset   = sizeof(struct ibv_tmh)
-                                    - sizeof(uct_rc_mlx5_hdr_t)
-                                    + iface->super.super.config.rx_headroom_offset;
-
     iface->tm.rndv_desc.super.cb  = uct_rc_mlx5_release_desc;
+
+    if (UCT_RC_MLX5_MP_ENABLED(iface)) {
+        iface->tm.eager_desc.offset = sizeof(struct ibv_tmh) +
+                                      iface->super.super.config.rx_headroom_offset;
+        iface->tm.am_desc.offset    = sizeof(uct_rc_mlx5_hdr_t) +
+                                      iface->super.super.config.rx_headroom_offset;
+        status = uct_iface_mpool_init(&iface->super.super.super,
+                                      &iface->tm.mp.tx_mp,
+                                      sizeof(uct_rc_iface_send_desc_t) +
+                                      UCT_RC_MLX5_TAG_BCOPY_MAX,
+                                      sizeof(uct_rc_iface_send_desc_t),
+                                      UCS_SYS_CACHE_LINE_SIZE,
+                                      &config->super.tx.mp,
+                                      iface->super.config.tx_qp_len,
+                                      uct_rc_iface_send_desc_init,
+                                      "tag_eager_send_desc");
+        if (status != UCS_OK) {
+            return;
+        }
+        iface->tm.bcopy_mp  = &iface->tm.mp.tx_mp;
+        iface->tm.max_zcopy = uct_ib_iface_port_attr(&iface->super.super)->max_msg_sz;
+        iface->tm.max_bcopy = UCT_RC_MLX5_TAG_BCOPY_MAX;
```

Comment:
max_bcopy is limited by the specific TX memory pool.
probably do not have a test witt max zcopy. Will limit to bcopy size as well

Reply:
but we create the TX mempool base on constant UCT_RC_MLX5_TAG_BCOPY_MAX on line 670, right?
can we calculate based on MTU+num strides instead of using this constant?

Reply:
one message can take several WQEs (that's the main purpose of this feature).
We can calculate a threshold based on seg_size+num_strides+num_wqes, but constant looks simpler

Reply:
i think it's better to calc (mayb in separate function if it's complicated), to reduce excessive memory consumption

Reply:
calculate based on what? We want to be able to send relatively big messages with tag eager protocol.
Currently we configure seg_size to be 48K when TM is enabled. Now, with these changes, we will maintain separate TX pool for sending big messages via tag protocol. RX pool will consist of MTU size segments. 
It is not reasonable to make TX pool less than 48K and according to measurements we need to be able to send >=64K messages eagerly. RX side should be able to handle that (the worst case it would go to RNR flow for some time). Maybe use some value slightly bigger than 64K (65K for instance)?